process/block/shardblock.go

```diff
@@ -739,7 +739,8 @@ func (sp *shardProcessor) CommitBlock(
 			continue
 		}
 
-		errNotCritical := sp.accounts.PruneTrie(rootHash)
+		// TODO add integration test to see that pruning works as expected
+		errNotCritical = sp.accounts.PruneTrie(rootHash)
```

Comment:
I suspect that you have to change to prevBlockHeader(finalHeaders[i]).GetRootHash() here

Reply:
Done.

Reply:
need to take it from storage instead of pool